emails/tests/views_tests.py

```diff
@@ -130,38 +132,42 @@ def setUp(self):
         send_raw_email_patcher.start()
         self.addCleanup(send_raw_email_patcher.stop)
 
-    def get_headers_from_mock_send_raw_email(self) -> dict[str, str]:
-        """Get headers from the message sent by mocked ses_client.send_raw_email"""
+    def get_details_from_mock_send_raw_email(self) -> tuple[str, str, dict[str, str]]:
+        """
+        Get sender, recipient, and headers from the message sent by mocked
+        ses_client.send_raw_email
+        """
         self.mock_send_raw_email.assert_called_once()
+        source = self.mock_send_raw_email.call_args[1]["Source"]
+        destinations = self.mock_send_raw_email.call_args[1]["Destinations"]
+        assert len(destinations) == 1
         raw_message = self.mock_send_raw_email.call_args[1]["RawMessage"]["Data"]
         headers: dict[str, str] = {}
         for line in raw_message.splitlines():
             if not line:
                 # Start of message body, done with headers
-                return headers
+                return source, destinations[0], headers
             assert ": " in line
             key, val = line.split(": ", 1)
             assert key not in headers
             headers[key] = val
         raise Exception("Never found message body!")
 
-    def test_single_recipient_sns_notification(self) -> None:
+    def test_single_recipient_sns_notification_legacy_headers(self) -> None:
```

Comment:
These changes are harder to read than I'd like (I'm tempted to open yet another cleanup PR). The pattern is an existing test, with `_legacy_headers` added to the name, that tests the headers without the new flag. The new test, with the old name, tests the headers with the new flag. When we remove the flag, we remove the `_legacy_headers` tests and the `@override_flag` decorators.